File: Leet-Codes/removeDoublesArrange.py
# 1. Leitura de quantos números serão inseridos
numbers = int(input("Digite quantos números quer escrever: "))

numbers_list = []
numbers_removed = []

# 2. Coleta dos números e separação de duplicatas
for terms in range(numbers):
    n = float(input("Digite seus números: "))
    
    if n in numbers_list:
        numbers_removed.append(n)
    else:
        numbers_list.append(n)

# 3. Quantidade de números únicos
quantity = len(numbers_list)

# 4. Ordenação usando Bubble Sort
for o in range(quantity):
    for p in range(quantity - 1):
        if numbers_list[p] > numbers_list[p + 1]:
            # Troca de valores
            save_value = numbers_list[p]
            numbers_list[p] = numbers_list[p + 1]
            numbers_list[p + 1] = save_value

# 5. Exibição do resultado final
print(f"Sua lista de números é: {numbers_list}. Duplicatas removidas: {numbers_removed}")

File: Leet-Codes/removeDoublesArrange.js
var numbers = parseInt(prompt("digite quantos numeros quer escrever: "));
var numbersList = [];
var numbersRemoved = [];

for(var terms = 0; terms < numbers; terms++){
    var n = parseFloat(prompt("digite seus numeros: "));
    if (numbersList.includes(n)){
        numbersRemoved.push(n);
    }
    else {
        numbersList.push(n);
    }
}

// CORREÇÃO 1: "lenght" estava escrito errado (o correto é "length")
var quantity = numbersList.length;

// CORREÇÃO 2: Ajuste no segundo laço (usando 'p' em vez de 'o') 
// e no limite da contagem ('quantity - 1')
for(var o = 0; o < quantity; o++){
    for(var p = 0; p < quantity - 1; p++){
        if(numbersList[p] > numbersList[p + 1]) {
            var saveValue = numbersList[p];
            numbersList[p] = numbersList[p + 1];
            numbersList[p + 1] = saveValue;
        }
    }
}

alert("Sua lista de numeros eh: " + numbersList + "." + " Duplicatas removidas: " + numbersRemoved);